eras/conway/impl/src/Cardano/Ledger/Conway/Rules/Deleg.hs

```diff
@@ -136,31 +164,42 @@ conwayDelegTransition = do
       checkStakeKeyNotAlreadyRegistered stakeCred dsUnified
       pure $ dState {dsUnified = acceptDepositForStakeKey stakeCred dsUnified ppKeyDeposit}
     ConwayUnRegCert stakeCred sMayDeposit -> do
-      checkStakeKeyIsAlreadyRegistered stakeCred dsUnified
-      checkStakeKeyHasZeroBalance stakeCred dsUnified
+      checkStakeKeyIsRegistered stakeCred dsUnified
+      checkStakeKeyHasZeroBalanceReward stakeCred dsUnified
       forM_ sMayDeposit $ checkDepositAgainstPaidDeposit stakeCred dsUnified
       pure $
         dState
-          { dsUnified = UM.domDeleteAll (Set.singleton stakeCred) dsUnified
+          { dsUnified = UM.domDeleteAll (Set.singleton stakeCred) dsUnified -- QUESTION: @WhatisRT: We auto-un-register him as a DRep?
           }
     ConwayDelegCert stakeCred delegatee -> do
-      checkStakeKeyIsAlreadyRegistered stakeCred dsUnified
+      checkStakeKeyIsRegistered stakeCred dsUnified
+      -- QUESTION: @WhatisRT: One cannot delegate if nothing is staked? This seems wrong because `processDelegation` also
+      -- processes stake delegation. What does it exactly mean to have a stake key present in the domain of the UMap?
+      -- It can mean any of the following exclusively and inclusively, and there is no inclusive implication here.
+      -- 1. that there is reward and deposit pait (could be zero?)
+      -- 2. that there is a staking mapped to a pool
+      -- 3. that there is a delegation of vote to a drep
       pure $
         dState
           { dsUnified = processDelegation stakeCred delegatee dsUnified
           }
     ConwayRegDelegCert stakeCred delegatee deposit -> do
-      deposit == ppKeyDeposit ?! IncorrectDepositDELEG deposit
-      checkStakeKeyNotAlreadyRegistered stakeCred dsUnified
+      checkDepositAgainstPParams ppKeyDeposit deposit
+      checkStakeKeyNotAlreadyRegistered stakeCred dsUnified -- QUESTION: Same here as above.
       pure $
         dState
           { dsUnified =
               processDelegation stakeCred delegatee $
                 acceptDepositForStakeKey stakeCred dsUnified ppKeyDeposit
           }
   where
-    acceptDepositForStakeKey stakeCred dsUnified ppKeyDeposit =
-      UM.RewDepUView dsUnified UM.∪ (stakeCred, UM.RDPair (UM.CompactCoin 0) (UM.compactCoinOrError ppKeyDeposit))
+    -- QUESTION: @WhatisRT: Why don't we track DRep deposits in VState like we track SPool deposits in DState,
```

Comment:
They should be tracked somewhere, there's a big map for them in the spec but I don't know where they'll end up in the code.

Reply:
Opened https://github.com/input-output-hk/cardano-ledger/issues/3638 to track this.